src/main/java/org/bosch/intern/core/EngineImpl.java

```
@@ -0,0 +1,140 @@
+package org.bosch.intern.core;
+
+import org.bosch.intern.Server.Server;
+import org.bosch.intern.entity.Author;
+import org.bosch.intern.entity.Book;
+import org.bosch.intern.exception.BookStoreException;
+import org.bosch.intern.service.BookStoreService;
+import org.bosch.intern.util.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class EngineImpl {
+    private BookStoreService bookStoreService;
+    private static final String EXIT = "You exit the program.";
+
+    public void run() {
+        bookStoreService = new BookStoreService();
+        boolean exitCommand = false;
+        try (Server server = new Server()) {
+            server.starServer(6999);
+            while (true) {
+                try {
+                    List<List<String>> response = new ArrayList<>();
+                    ClientRequest clientRequest = server.readRequest();
+                    switch (clientRequest.getMethod()) {
+                        case READ:
+                            response.add(readEntity(clientRequest));
+                            break;
+                        case ADD:
+                            response.add(addEntity(clientRequest));
+                            break;
+                        case READALL:
+                            response = readAllEntity(clientRequest);
+                            break;
+                        case Exit:
+                            exitCommand = true;
+                            break;
+                        default:
+                            List<String> defaultList = new ArrayList<>();
+                            defaultList.add(ConstantMessages.INVALID_COMMAND);
```

You can throw an exception here. BookStoreException, it will be caught in the catch bellow. This will be one step closer to do it cleaner